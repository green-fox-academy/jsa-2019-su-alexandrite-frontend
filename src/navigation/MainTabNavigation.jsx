import React from 'react';
import { Image } from 'react-native';
import { createAppContainer } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Watchlists from '../watchlists';
import Investments from '../investments';
import Account from '../account';
import Playground from '../playground';

import accountIcon from '../../assets/icons/bottom-tab/me.png';
import watchlistsIcon from '../../assets/icons/bottom-tab/watchlists.png';
import investmentsIconActive from '../../assets/icons/bottom-tab/investments-active.png';
import StockDetails from '../stock-details';
import Search from '../search';

const InvestmentsStack = createStackNavigator({
  Investments: {
    screen: Investments,
    path: 'investments/',
  },
  Search,
});

const WatchlistsStack = createStackNavigator({
  Watchlists: {
    screen: Watchlists,
    path: 'watchlists/',
  },
  Search,
});

const AccountStack = createStackNavigator({
  Account: {
    screen: Account,
    path: 'account/',
  },
});

const PlaygroundStack = createStackNavigator({
  Playground,
  StockDetails,
  Search,
});

const renderIcon = (icon, tintColor) => (<Image source={icon} style={{ tintColor }} />);

const tabs = {
  ...(process.env.NODE_ENV === 'development' && { Playground: PlaygroundStack }),
  Investments: {
    screen: InvestmentsStack,
    navigationOptions: {
      tabBarIcon: ({ tintColor }) => renderIcon(investmentsIconActive, tintColor),
    },
  },
  Watchlists: {
    screen: WatchlistsStack,
    navigationOptions: {
      tabBarIcon: ({ tintColor }) => renderIcon(watchlistsIcon, tintColor),
    },
  },
  Me: {
    screen: AccountStack,
    navigationOptions: {
      tabBarIcon: ({ tintColor }) => renderIcon(accountIcon, tintColor),
    },
  },
};

const BottomTabNavigatorConfig = {
  tabBarOptions: {
    activeTintColor: '#566ed3',
    inactiveTintColor: '#999999',
  },
};

const TabNavigator = createBottomTabNavigator(tabs, BottomTabNavigatorConfig);
export default createAppContainer(TabNavigator);
