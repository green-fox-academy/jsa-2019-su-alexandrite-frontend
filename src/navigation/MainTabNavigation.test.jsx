import renderer from 'react-test-renderer';
import { renderIcon } from './MainTabNavigation';
import investmentsIconActive from '../../assets/icons/bottom-tab/investments-active.png';

describe('<MainTabNavigation />', () => {
  it('renderIcon Should render and match the snapshot', () => {
    const tree = renderer.create(
      renderIcon(investmentsIconActive, '#eee'),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
