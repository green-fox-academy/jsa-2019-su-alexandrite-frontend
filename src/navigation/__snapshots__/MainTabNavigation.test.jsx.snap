// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainTabNavigation /> renderIcon Should render and match the snapshot 1`] = `
<Image
  source={1}
  style={
    Object {
      "tintColor": "#eee",
    }
  }
/>
`;
